### ISSUE TITLE
Remove multiplied Characteristics when reading rows with the same SourceName

### ISSUE THREAD

**Issue by hcwi** [ISA-tools/isa-api#269]
In converting tab to json, having a few rows with the same Source Name leads to creating a long vector of Characteristics with all of the values from all rows

**Comment by djcomlab** [comment]
Hi @hcwi,

Please can you provide an example ISA-Tab and that this problem occurs with and the line of code you write to execute the `isatab2json` converter?

When I test against our test data (BII studies in https://github.com/ISA-tools/ISAdatasets/) I cannot replicate the reported behaviour.

**Comment by hcwi** [comment]
Ok, sent via email.

**Comment by djcomlab** [comment]
Can you confirm that you're expecting 11 characteristic values per source, correct?

**Comment by hcwi** [comment]
Actually 10, as there are 10 Characteristics columns in tab file. And currently I am getting 60 (10 sources * 6 treatments)

**Comment by djcomlab** [comment]
OK, that's strange... I can confirm the 60, but my fix is coming out with 11...

**Comment by djcomlab** [comment]
Oh actually, I'm finding a Geographic location is coming up twice for one source now...

**Comment by djcomlab** [comment]
I see there are two different term accessions used for the same Geographic location values. Is this intended? 
```
Characteristics[Geographic location], Term Source REF, Term Accession
"Antoniny, Leszno, Poland", GAZ, http://purl.obolibrary.org/obo/GAZ_00022543
"Antoniny, Leszno, Poland", GAZ, http://purl.obolibrary.org/obo/GAZ_00022648
```

**Comment by hcwi** [comment]
Rather not, probably just a typo, but I'll need to check the accessions. Thanks!

What is the expected outcome in the situation of different values in some column? Is the Characteristic[x] added twice with both values? Or is it assumed that the sources are different entities, despite the same name?

**Comment by djcomlab** [comment]
Actually it shouldn't allow this and either overwrite the value on the most recent write, or raise an error...

**Comment by djcomlab** [comment]
Fixed in 5b3c5e21bcc98ec40ec00236ec75b4f5dc500143
